### PR TITLE
enhance: remove the rpc level of coordinator when enabling standalone or mixcoord

### PR DESCRIPTION
**Pull request by chyezh**
issue: #33285
pr: #37815

- move most cgo opeartions related to search/query into segcore package for reusing for streamingnode.
- add go unittest for segcore operations.

### CODE REVIEW

File: configs/milvus.yaml [hunk 1]
```diff
@@ -846,6 +846,7 @@ common:
   usePartitionKeyAsClusteringKey: false # if true, do clustering compaction and segment prune on partition key field
   useVectorAsClusteringKey: false # if true, do clustering compaction and segment prune on vector field
   enableVectorClusteringKey: false # if true, enable vector clustering key and vector clustering compaction
+  localRPCEnabled: false # enable local rpc for internal communication when mix or standalone mode.
```
Comment (czs007) [inline configs/milvus.yaml:849]: It would be preferable to change the default behavior to True.

**Comment by chyezh** [comment]
hold

**Pull request closed without merging**